### PR TITLE
Finalize editing when hitting the enter key on a single line TextField

### PR DESCRIPTION
**Pull request by franciscojma86**
Behavior remains the same for multiline TextFields. This behavior already happens on iOS and Android, but not desktop. #23014

### CODE REVIEW

File: packages/flutter/lib/src/widgets/editable_text.dart [hunk 1]
```diff
@@ -511,44 +511,43 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
...
+      widget.onEditingComplete();
+    } else {
+      // Default behavior if the developer did not provide an
+      // onEditingComplete callback: Finalize editing and remove focus.
+      widget.controller.clearComposing();
+      if (shouldUnfocus)
+        widget.focusNode.unfocus();
+    }
+
+        // Invoke optional callback with the user's submitted content.
```
Comment (gspencergoog) [inline packages/flutter/lib/src/widgets/editable_text.dart]: Indenting is wrong on this comment.

File: packages/flutter/lib/src/widgets/editable_text.dart [hunk 2]
```diff
@@ -511,44 +511,43 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
   void performAction(TextInputAction action) {
     switch (action) {
       case TextInputAction.newline:
-        // Do nothing for a "newline" action: the newline is already inserted.
+        // If this is a multiline EditableText, do nothing for a "newline"
+        // action; The newline is already inserted. Otherwise, finalize
+        // editing.
+        if (widget.maxLines == 1)
+          _finalizeEditing(true);
```
Comment (gspencergoog) [inline packages/flutter/lib/src/widgets/editable_text.dart]: Can you please add a test for this change in behavior?

**Review by gspencergoog — CHANGES_REQUESTED** [review]

File: packages/flutter/lib/src/widgets/editable_text.dart (hunk 2 above)
Comment (franciscojma86) [reply packages/flutter/lib/src/widgets/editable_text.dart]: Done

**Comment by franciscojma86** [comment]
All comments addressed and checks pass :) PTAL!

File: packages/flutter/test/widgets/editable_text_test.dart [hunk 3]
```diff
@@ -745,6 +745,95 @@ void main() {
...
+        key: editableTextKey,
+        controller: TextEditingController(),
+        focusNode: focusNode,
+        style: Typography(platform: TargetPlatform.android).black.subhead,
+        cursorColor: Colors.blue,
+        maxLines: 1,
+        onEditingComplete: () {
+          onEditingCompleteCalled = true;
+          assert(!onSubmittedCalled);
+
```
Comment (gspencergoog) [inline packages/flutter/test/widgets/editable_text_test.dart]: Remove extra blank line.

**Review by gspencergoog — APPROVED** [review]

**Comment by gspencergoog** [comment]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

(modulo removing the blank line)

File: packages/flutter/test/widgets/editable_text_test.dart (hunk 3 above)
Comment (franciscojma86) [reply packages/flutter/test/widgets/editable_text_test.dart]: Done

**Comment by gspencergoog** [comment]
OK, great!  Thanks for the PR.  Will land when everything's green.

**Pull request merged**